spec/comm.mediawiki

```diff
@@ -182,14 +182,24 @@ connection is established, the client will supply their account ID and signature
 
 If a client unexpectedly disconnects with active orders, the orders may match in
 the client's absence. A list of any pending matches is included in the response.
+If the client has a broken rule, they will not be able to trade for the duration
+of their penalization.
 
 <code>result</code>
 {|
 ! field   !! type   !! description
 |-
-| matches || &#91;object&#93; || list of [[orders.mediawiki/#Match_negotiation|Match objects]]
+| matches      || &#91;object&#93; || list of [[orders.mediawiki/#Match_negotiation|Match objects]]
 |-
-| sig     || string || hex-encoded server's signature of the serialized connection data
+| broken rule  || int || the client's current [[community.mediawiki/#Rules_of_Community_Conduct|rule status]]
+|-
+| ban time     || int || the UNIX time in milliseconds at which the penalization occured if applicable
+|-
+| ban duration || int || the duration in milliseconds of the penalization if applicable
+|-
+| ban details  || string || a message describing the penalization if applicable
+|-
+| sig          || string || hex-encoded server's signature of the serialized connection data
```

Comment:
In the discussion about sending the notification on reconnect, that will already happen automatically if we use `SendWhenConnected` to send the notification, but that has a connect timeout that eventually expires the message, so it makes sense to include the data in the `'connect'` response.

However, this is a bunch of data that is not going to be set for users in good standing, so it will have to be omitted if there are no violations.  Although I suppose in a strikes system it would be helpful to give the user their strike count.

Also, while the notifications are communicating an event, `'connect'` is sending a state, so fields like `broken rule` have a different interpretation here.  (was it the rule that caused the account to reach the strike limit in a sequence of different rule violations?)  This could maybe be solved by revising that field's description to indicate that it is the most recently violated rule or the one that caused the suspension.

Lastly, I'm not sure about the "ban" wording.  The user can presently still connect to complete active swaps, but depending on the nature of the ban/suspension they just won't be able to place new orders and their booked orders were probably unbooked.

Reply:
I agree with all points. Hopefully the wording is better now too.